website/docs/docs/ai/using.md

```diff
@@ -0,0 +1,251 @@
+      .get(this.withPath("/product/" + id), {
+        headers: {
+          Authorization: this.generateAuthToken(),
+        },
+      })
+      .then((r) => new Product(r.data));
+  }
+}
+```
+
+...and the following `Product` definition:
+
+**product.js**
+```js
+export class Product {
+  constructor({id, name, price}) {
+    this.id = id
+    this.name = name
+    this.price = price
+  }
+}
+```
+
+We can generate pact tests using the `pactflow-ai generate code` subcommand:
+
+```sh
+pactflow-ai generate code \
+  --output ./src/api.pact.spec.ts \
+  --language typescript \
+  ./src/api.js \
+  ./src/product.js
+```
+
+Note how we can pass in multiple files to the command. Providing the right information to PactFlow AI will ensure it has the context to generate usable output.
+
+#### Generating Tests from your OpenAPI Descriptions + Code
+
+If you have both an OAD and code, you can pass both to the `openai` subcommand as follows:
+
+```sh
+pactflow-ai generate openapi \
+  --spec ./products.yml \
+  --endpoint "/product/{id}" \
+  --output ./src/api.pact.spec.ts \
+  --language typescript \
+  --code ./src/product.js \
+  --code ./src/api.js
+ ```
+
+This tends to provide the best output, as it has the most information about how the API works and how your code integrates with it.
+
+#### Generating Tests from Request-Response
+
+In some cases, you may not have access to an OAD or code. You may already have a large test suite where you have - or can - record the request and response details. You may simply have detailed descriptions such as a BDD suite, describing how the solution should work.
+
+In these cases, you can create a suite of Pact tests from these recordings.
+
+Given the following request-response pairs (taken from the output of `curl` command, and in the format of an [HTTP Message](https://developer.mozilla.org/en-US/docs/Web/HTTP/Messages)):
+
+**get.request.http**
+```
+GET /product/10 HTTP/1.1
+Host: api.example.com
+User-Agent: curl/8.1.2
+Authorization: Bearer notarealtoken
+Accept: application/json; charset=UTF-8
+```
+
+**get.response.http**
+```
+HTTP/1.1 200 OK
+Content-Type: application/json; charset=utf-8
+Content-Length: 96
+
+{
+  "id": "10",
+  "name": "Aussie",
+  "type": "Pizza",
+  "version": "v1",
+  "price": 9.99,
+}
+```
+
+Run the following command to generate a Pact test:
+
+```sh
+ pactflow-ai generate request-response \   # use the request-response subcommand
+  --request ./capture/get.request.http \   # path to the request description
+  --response ./capture/get.response.http \ # path to the response description
+  --language typescript \
+  --output ./src/api.pact.spec.ts
+```
+
+**Considerations**
+
+As this form of generation does not have access to the client source code, tests generated will use a placeholder API client in place of the real API client, when calling the mock service in the test.
+
+Be sure to update your tests to insert the real API client to ensure the tests are valid.
```

Comment:
Once the unified endpoint/refactor we're doing is done, we'll be able to combine request/response with all other inputs.

One consideration about the implementation will be that the request-response is deemed to be _most specific_ and will drive what Pact test is going to be generated. For example, if combining a request-response with an OpenAPI doc, the AI will automatically determine what part of the OAD is relevant and trim the rest (in detail, it builds its own `--endpoint` matcher)